File: src/models/milestone.ts
export type MilestoneMemo = {
  id: number;
  slug_title: string;
  title: string;
};

export type Milestone = {
  id: number;
  workspace_id: number;
  date: string;
  title: string;
  memos: MilestoneMemo[];
  completed_at?: string | null;
  created_at: string;
  updated_at: string;
};

File: src-tauri/src/models/milestone.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct MilestoneMemo {
    pub id: i32,
    pub slug_title: String,
    pub title: String,
}

#[derive(Serialize, Deserialize)]
pub struct Milestone {
    pub id: i32,
    pub workspace_id: i32,
    pub date: String,
    pub title: String,
    pub memos: Vec<MilestoneMemo>,
    pub completed_at: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}
